linera-base/src/policy.rs

```diff
@@ -6,17 +6,25 @@
 use std::{collections::BTreeSet, fmt};
 
 use async_graphql::InputObject;
-use linera_base::{
-    data_types::{Amount, ArithmeticError, BlobContent, CompressedBytecode, Resources},
-    ensure,
-    identifiers::BlobType,
-};
+use linera_witty::{WitLoad, WitStore, WitType};
 use serde::{Deserialize, Serialize};
 
-use crate::ExecutionError;
+use crate::data_types::{Amount, ArithmeticError, Resources};
 
 /// A collection of prices and limits associated with block execution.
-#[derive(Eq, PartialEq, Hash, Clone, Debug, Serialize, Deserialize, InputObject)]
+#[derive(
+    Eq,
+    PartialEq,
+    Hash,
+    Clone,
+    Debug,
+    Serialize,
+    Deserialize,
+    InputObject,
+    WitLoad,
+    WitStore,
+    WitType,
+)]
 pub struct ResourceControlPolicy {
```

Comment:
Do we need to expose the entire vector `ResourceControlPolicy` to the SDK?

Reply:
What exactly are EVM contracts trying to achieve?

Reply:
I think that smart contracts should be able to see the price of the gas when running. Smart contracts need to know the cost of their operation and decide based on that. So, I do not see the harm in providing too much information.

Reply:
The data that should be provided to the run is the gas price in the struct `TxEnv`.

Right now, that cost is not available in the `ResourceControlPolicy`. It should be after the PR https://github.com/linera-io/linera-protocol/pull/3715